components/feature/addons/src/main/java/mozilla/components/feature/addons/AddOn.kt

```diff
@@ -72,4 +72,45 @@ data class AddOn(
         val average: Float,
         val reviews: Int
     ) : Parcelable
+
+    /**
+     * Returns a list of id resources per each item on the [permissions] list.
+     */
+    @Suppress("ComplexMethod")
+    fun translatePermissions(): List<Int> {
+        val permissionIds = mutableListOf<Int>()
+        permissions.forEach { permission ->
```

Comment:
Instead of a `mutableList`, you can just use `val permissionIds = permissions.mapNotNull { }`. Then you can actually just `return permissions.mapNotNull { }`

Reply:
Yes, we can. I was doing a micro optimization. To not populate the list with null items and then have to filter for null values. I think the reliability is better, I will change it 👍

Reply:
OK yeah, there won't be a perf. difference. If it all the static map will be more efficient...

Reply:
👍

Reply:
Updated!